### DISCUSSION TITLE
Adding products based on external APIs

### DISCUSSION THREAD

**Discussion by henninghorn** [medusajs/medusa#8238] [Q&A]
I'm building a platform where I need to add products based on responses from external APIs, e.g. hotel rooms. Instead of creating my own shop/cart system, I'm considering using Medusa for handling carts/orders etc.

But I'm not sure if and how this is possible with Medusa 🤞

Let's say I search for a hotel room for 25/3 to 28/3.
My external API return two available rooms, room A which costs $100 and room B which costs $200.

How would you add that item to the cart?
I know I can add metadata to the item I'm adding to the cart.
But how can I control the price of the item?

And furthermore, how would you trigger the final booking request when payment has been completed?

I'd love to get any pointers to documentation, examples or similar.
Thanks!

**Comment by Hermitauge** [comment]
I have the same question. Please someone respond to this, would really appreciate it.
